src/sentry/api/endpoints/project_metrics_extraction_rules.py

```diff
@@ -104,22 +108,27 @@ def post(self, request: Request, project: Project) -> Response:
                     configs.append(
                         SpanAttributeExtractionRuleConfig.from_dict(obj, request.user.id, project)
                     )
+
+                validate_number_of_extracted_metrics(project)
+
                 schedule_invalidate_project_config(
                     project_id=project.id, trigger="span_attribute_extraction_configs"
                 )
 
-        except Exception:
-            sentry_sdk.capture_exception()
-            return Response(
-                status=400,
+            persisted_config = serialize(
+                configs,
+                request.user,
+                SpanAttributeExtractionRuleConfigSerializer(),
             )
+            return Response(data=persisted_config, status=200)
 
-        persisted_config = serialize(
-            configs,
-            request.user,
-            SpanAttributeExtractionRuleConfigSerializer(),
-        )
-        return Response(data=persisted_config, status=200)
+        except MetricsExtractionRuleValidationError as e:
+            logger.warning("Failed to update extraction rule", exc_info=True)
+            return Response(status=400, data={"detail": str(e)})
+
+        except Exception:
+            logger.exception("Failed to update extraction rule")
```

Comment:
This is nit, but if some general Exception happens that we are not aware of, I would also log the exception (maybe even in Sentry) so that we can act on it if needed